proto/message_contents/proof.proto

```diff
@@ -0,0 +1,46 @@
+// Proof contains signature-based bindings for different use cases.
+syntax = "proto3";
+
+package xmtp.message_contents;
+
+import "message_contents/signature.proto";
+
+option go_package = "github.com/xmtp/proto/v3/go/message_contents";
+option java_package = "org.xmtp.proto.message.contents";
+
+// An association proof proves a child key (and/or other data) has been
+// associated with a parent key.
+message AssociationProof {
+    // Legacy: "XMTP : Create Identity ..." format
+    message LegacyCreateIdentityProof {
```

Comment:
DUST: Legacy will have less and less meaning and time moves on. 

Rename to 'CreateIdentityProof' or 'StaticCreateIdentityProof' ?

Reply:
This is a great point